integ/components/deadline/deadline_01_repository/lib/repository-testing-tier.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path';
+import { ILogGroup } from '@aws-cdk/aws-logs';
+import { CfnOutput, Construct } from '@aws-cdk/core';
+import { MongoDbInstaller, MongoDbSsplLicenseAcceptance, MongoDbVersion } from 'aws-rfdk';
+import { Repository } from 'aws-rfdk/deadline';
+import { StorageStruct } from '../../../../lib/storage-struct';
+import { TestingTier, TestingTierProps } from '../../../../lib/testing-tier';
+
+/**
+ * Interface for RepositoryTestingTier properties
+ */
+export interface RepositoryTestingTierProps extends TestingTierProps {
+  /**
+   * Array of StorageStructs representing different test cases
+   */
+  readonly structs: Array<StorageStruct>;
+}
+
+/**
+ * Testing Tier for the Deadline Repository integration test
+ *
+ * Creates a test bastion and configures it to connect to one or more Deadline Repository constructs for testing
+ *
+ * Resources Deployed
+ * ------------------------
+ * - A BastionLinuxHost instance
+ *
+ * Security Considerations
+ * ------------------------
+ * - The bastion instance created by this test is configured to access farm resources on their default ports
+ *   Test scripts stored on the bastion are used to read/write to the repository database and the file system.
+ *   At execution the tests retrieve the value of secrets for the database password and authentication cert.
+ */
+export class RepositoryTestingTier extends TestingTier {
+  constructor(scope: Construct, id: string, props: RepositoryTestingTierProps) {
+    super(scope, id, props);
+
+    const structs = props.structs;
+    structs.forEach( storageStruct => {
+
+      const testSuiteId = 'DL' + (structs.indexOf(storageStruct) + 1).toString();
+
+      const repo = storageStruct.repo;
+      this.configureRepo(testSuiteId, repo);
+
+      const database = storageStruct.database;
+      this.configureDatabase(testSuiteId, database);
+
+      const efs = storageStruct.efs;
+      this.testInstance.connections.allowToDefaultPort(efs);
+
+      const cert = storageStruct.database.cert;
+      this.configureCert(testSuiteId, cert);
+    });
+
+    this.configureBastionUserData({
+      testingScriptPath: path.join(__dirname, '../scripts/bastion/testing'),
+    });
+    this.fetchDocdbCert();
+    this.installMongodb();
+
+  }
+
+  /**
+   * Mounts the Deadline repository's file system to the bastion and outputs the name of its log group
+   *
+   * @param testSuiteId Test case to configure the repository for
+   * @param repo Repository object to connect to the test Bastion
+   */
+  private configureRepo(testSuiteId:string, repo:Repository) {
```

Comment:
Fixed in all the cases I could find. If I missed any I can fix them in an upcoming commit that will revise the commenting in a bunch of these files.

Reply:
If we want to match the style of the rest of the codebase ([one example](https://github.com/aws/aws-rfdk/blob/cb68992de1c72f7997de6ff81f1b0c09e88dacaf/packages/aws-rfdk/lib/deadline/lib/render-queue.ts#L234)), we'd go with a keeping a space between the `:` and the type, so:

```ts
private configureRepo(testSuiteId: string, repo: Repository) {
```

in this case. It looks like you've pushed a change to remove the space everywhere 🤦 . If you do want to do this, we don't need to run another round of end-to-end testing. It would be sufficient to make sure the code builds properly. If you want to defer this clean-up to get this merged, I'm also okay with that. What do you think?

Reply:
Added #132 to track this